rs/machine/state.py

```diff
@@ -200,3 +200,22 @@ def extract_card_from_text(text):
         for idx, choice in enumerate(options):
             options[idx] = choice.replace("+", "")
         return options
+
+    def get_act_4_keys(self):
+        # just the first letter of the key is enough
+        keys = []
+        if "keys" not in self.game_state():
+            return ['Communication Mod out of date']
```

Comment:
Will this cause a nice readable error to show up in the STS Mod or logs? We should probably be logging it ourselves as well in the bottled_ai logs for more visibility.

Reply:
It will not, and OK sounds good!

Reply:
After further discussion: I've removed the message from state.py and isolated that handling to logger.py